src/main/java/frc/robot/subsystems/BallPath/Elevator/ElevatorImpl.java

```diff
@@ -25,6 +29,10 @@
     private volatile ElevatorAction action = ElevatorAction.NONE;
     private boolean lastPresent = false;
     private final Queue<Double> sensorSamples;
+    private final I2C.Port i2cPort = I2C.Port.kOnboard;
+    private final ColorSensorV3 elevatorColorSensor = new ColorSensorV3(i2cPort);
```

Comment:
This is a repeated initialization of a new sensor object representing the same physical sensor on the same physical port - ie there is already a `new ColorSensorV3(I2C.Port.kOnboard)` in Robot.java. It should be instantiated only once in `Robot.java` and passed into here as a constructor parameter.